sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/WebApplicationFirewallPolicyImpl.java

```diff
@@ -0,0 +1,252 @@
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of the {@link WebApplicationFirewallPolicy} interface.
+ */
+public class WebApplicationFirewallPolicyImpl extends GroupableResourceImpl<
+    WebApplicationFirewallPolicy, WebApplicationFirewallPolicyInner, WebApplicationFirewallPolicyImpl, NetworkManager>
+    implements WebApplicationFirewallPolicy,
+    WebApplicationFirewallPolicy.Definition,
+    WebApplicationFirewallPolicy.Update,
+    WebApplicationFirewallPolicy.UpdateStages.WithRequestBodyOrUpdate {
+    private static final String BOT_DETECTION_RULE_SET_TYPE = "Microsoft_BotManagerRuleSet";
+    private static final String BOT_DETECTION_RULE_SET_VERSION_DEFAULT = "0.1";
+    private static final String DEFAULT_MANAGED_RULE_SET = "OWASP";
+    private static final String DEFAULT_MANAGED_RULE_SET_VERSION = "3.2";
+
+    protected WebApplicationFirewallPolicyImpl(String name, WebApplicationFirewallPolicyInner innerObject, NetworkManager manager) {
+        super(name, innerObject, manager);
+    }
+
+    @Override
+    public WebApplicationFirewallMode mode() {
+        if (this.innerModel().policySettings() == null) {
+            return null;
+        }
+        return this.innerModel().policySettings().mode();
+    }
+
+    @Override
+    public boolean isRequestBodyInspectionEnabled() {
+        return ResourceManagerUtils.toPrimitiveBoolean(this.innerModel().policySettings().requestBodyCheck());
+    }
+
+    @Override
+    public Integer requestBodySizeLimitInKb() {
+        return this.innerModel().policySettings().requestBodyInspectLimitInKB();
+    }
+
+    @Override
+    public Integer fileUploadSizeLimitInMb() {
+        return innerModel().policySettings().fileUploadLimitInMb();
+    }
+
+    @Override
+    public PolicySettings getPolicySettings() {
+        return innerModel().policySettings();
+    }
+
+    @Override
+    public ManagedRulesDefinition getManagedRules() {
+        return innerModel().managedRules();
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return WebApplicationFirewallEnabledState.ENABLED.equals(innerModel().policySettings().state());
+    }
+
+    @Override
+    public List<String> getAssociatedApplicationGatewayIds() {
+        if (CoreUtils.isNullOrEmpty(this.innerModel().applicationGateways())) {
+            return Collections.emptyList();
+        }
+        return this.innerModel().applicationGateways()
+            .stream()
+            .map(ApplicationGatewayInner::id)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<ApplicationGateway> getAssociatedApplicationGateways() {
```

Comment:
not exactly related, guideline would recommend IterableStream
https://azure.github.io/azure-sdk/java_introduction.html#java-sync-client-name

We probably still use List for consistency.